src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -212,14 +234,50 @@ public void Write(int pinNumber, PinValue value)
             _driver.Write(logicalPinNumber, value);
         }
 
+        /// <summary>
+        /// Returns the currently set pin mode by directly reading the hardware
+        /// </summary>
+        /// <param name="pinNumber">Pin number</param>
+        /// <returns>(Alternate) Pin mode. 0 = Alt0, 1= Alt1... -1 Gpio Input, -2 Gpio Output</returns>
+        protected internal int GetAlternatePinMode(int pinNumber)
+        {
+            int logicalPinNumber = GetLogicalPinNumber(pinNumber, NumberingScheme);
+            // TODO: Maybe use an interface for this feature query instead?
+            RaspberryPi3Driver driver = _driver as RaspberryPi3Driver;
```

Comment:
no, it should be generic or implementation specific, this will be very annoying to support long term

Reply:
I was trying to keep the impact on the System.Device assembly as low as possible (by not adding new methods in the base classes), but I'm happy to do it right.